src/Symfony/Component/Lock/Store/ZookeeperStore.php

```diff
@@ -0,0 +1,239 @@
+        if (!$this->zookeeper->exists($lockNodePath)) {
+            return false;
+        }
+
+        return $this->zookeeper->get($lockNodePath) === $this->getToken($key);
+    }
+
+    /**
+     * {@inheritdoc}
+     *
+     * @throws NotSupportedException
+     */
+    public function waitAndSave(Key $key)
+    {
+        throw new NotSupportedException();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function putOffExpiration(Key $key, $ttl)
+    {
+    }
+
+    /**
+     * @param int    $errNo  Error number
+     * @param string $errStr Error String
+     *
+     * @throws LockConflictedException
+     * @throws LockAcquiringException
+     */
+    public static function errorHandlerForCreateNode($errNo, $errStr)
+    {
+        if (E_WARNING === $errNo && 'Zookeeper::create(): error: node exists' === $errStr) {
+            throw new LockConflictedException($errStr);
+        }
+
+        throw new LockAcquiringException($errStr);
+    }
+
+    /**
+     * Create a zookeeper node.
+     * If it has a path that doesn't exists, then this method creates the path with the same permissions before
+     * creating a node.
+     *
+     * @param string $node  The node which needs to be created
+     * @param string $value The value to be assigned to a zookeeper node
+     * @param array  $acl   Permissions settings on a zookeeper node
+     * @param int    $type  Type of zookeeper node OPTIONAL
+     *
+     * @return string Path of the node created
+     *
+     * @throws LockConflictedException
+     * @throws LockAcquiringException
+     */
+    private function createNewLock(
+        string $node,
+        string $value = '',
+        array $acl = array(),
+        int $type = Zookeeper::EPHEMERAL
+    ): string {
+        set_error_handler(array(self::class, 'errorHandlerForCreateNode'));
+        try {
+            // If the node exists then update the value of the node.
+            if ($this->zookeeper->exists($node)) {
+                throw new LockConflictedException('Lock already exists');
+            }
+
+            // Default Node Permissions
+            if (empty($acl)) {
+                $acl = array(array('perms' => Zookeeper::PERM_ALL, 'scheme' => 'world', 'id' => 'anyone'));
+            }
+
+            // Check if there is a path that needs to be created before creating a node.
+            $parts = array_filter(explode('/', $node));
+            $subpath = '';
+
+            // If a node needs to be created with the path "/node1/node2/node3",
+            // this loop will create "/node1" and "/node1/node2" before creation of deepest level node.
+            for ($numberOfParts = count($parts); $numberOfParts > 1; $numberOfParts = count($parts)) {
+                $subpath .= '/'.array_shift($parts);
+
+                if ($this->zookeeper->exists($subpath)) {
+                    continue;
+                }
+
+                $nodeCreated = $this->zookeeper->create($subpath, null, $acl);
```

Comment:
It would be easier, but I also wanted to provide the functionality of having intermediate nodes if anyone wanted to organize those. Ideally it would be preferred to have just `/node` for the sake of simplicity, but I can specify this in the documentation as a recommendation if that works for you.

Reply:
IMHO the client shouldn't have to know how stores works internally, and doesn't need to manipulate it. And I'm not sure that organizing "ephemeral nodes" worth it.
Moreover, the service asking a Lock may not (shouldn't?) be aware of the store used. It may not be aware using keys with slashes have a special meaning.

The issue with this feature are:
 - it add a lot of complexity to the method
 - it requires to create intermediate nodes that you can't delete theme afterward. what if a jobs locks `/foo/{id}/bar` and iterate over billions of `ids`?
 
Do you have a use case in mind?

Reply:
I was following the recipe from here: https://zookeeper.apache.org/doc/r3.1.2/recipes.html#sc_recipes_Locks

But that protocol was recommending `ephermeal + sequence` nodes. Since in our case we don't use sequence nodes, I think it is fine to not support multi-level nodes.

Changed the implementation to support single ephemeral node. In this case, I am validating the input during lock creation. Also added a test case for this.